java.lang.ExceptionInInitializerError when using alpaca-java inside of spring-boot application

When alpaca-java.jar is packaged inside of spring-boot application, AlpacaProperties class fails to initialize:

```
Caused by: java.lang.ExceptionInInitializerError
        at net.jacobpeterson.alpaca.rest.AlpacaClient.<init>(AlpacaClient.java:79)
        at net.jacobpeterson.alpaca.rest.AlpacaClient.<init>(AlpacaClient.java:51)
        at net.jacobpeterson.alpaca.AlpacaAPI.<init>(AlpacaAPI.java:149)
        ... 
Caused by: java.lang.IllegalArgumentException
        at net.jacobpeterson.alpaca.model.properties.EndpointAPIType.fromValue(EndpointAPIType.java:48)
        at net.jacobpeterson.alpaca.properties.AlpacaProperties.<clinit>(AlpacaProperties.java:30)
        ... 
```

Subsequent references to the AlpacaAPI fail with:

```
Caused by: java.lang.NoClassDefFoundError: Could not initialize class net.jacobpeterson.alpaca.properties.AlpacaProperties
        at net.jacobpeterson.alpaca.rest.AlpacaClient.<init>(AlpacaClient.java:79)
        at net.jacobpeterson.alpaca.rest.AlpacaClient.<init>(AlpacaClient.java:51)
        at net.jacobpeterson.alpaca.AlpacaAPI.<init>(AlpacaAPI.java:149)
```

**Rootcause:**

In case of spring-boot the libraries are loaded by application class loader, and not by system class loader.

Changing `ClassLoader.getSystemClassLoader()` to `PropertyUtil.class.getClassLoader()` addresses the issue.

--- a/src/main/java/net/jacobpeterson/alpaca/util/properties/PropertyUtil.java
+++ b/src/main/java/net/jacobpeterson/alpaca/util/properties/PropertyUtil.java
@@ -81,11 +81,14 @@ public class PropertyUtil {
      * @return the properties
      */
     public synchronized static Properties loadPropertyFile(String propertyFile, String defaultPropertyFile) {
-        Properties properties = null;
+        ClassLoader classLoader = PropertyUtil.class.getClassLoader();
+        if (classLoader == null) {
+            classLoader = ClassLoader.getSystemClassLoader();
+        }
 
         // Load the default property file if exists
         Properties defaultProperties = null;
-        InputStream defaultPropertyStream = getClassLoader().getResourceAsStream(defaultPropertyFile);
+        InputStream defaultPropertyStream = classLoader.getResourceAsStream(defaultPropertyFile);
 
         if (defaultPropertyStream != null) {
             defaultProperties = new Properties();
@@ -110,7 +113,8 @@ public class PropertyUtil {
         }
 
         // Load the property file
-        InputStream propertyStream = getClassLoader().getResourceAsStream(propertyFile);
+        Properties properties = null;
+        InputStream propertyStream = classLoader.getResourceAsStream(propertyFile);
 
         if (propertyStream != null) {
             // Add default properties if they were found
@@ -140,13 +144,5 @@ public class PropertyUtil {
         }
 
         return properties;
-    }
-
-    private static ClassLoader getClassLoader() {
-        ClassLoader classLoader = PropertyUtil.class.getClassLoader();
-        if (classLoader == null) {
-            classLoader = ClassLoader.getSystemClassLoader();
-        }
-        return classLoader;
     }
 }

--- a/src/main/java/net/jacobpeterson/alpaca/util/properties/PropertyUtil.java
+++ b/src/main/java/net/jacobpeterson/alpaca/util/properties/PropertyUtil.java
@@ -85,7 +85,7 @@ public class PropertyUtil {
 
         // Load the default property file if exists
         Properties defaultProperties = null;
-        InputStream defaultPropertyStream = PropertyUtil.class.getClassLoader().getResourceAsStream(defaultPropertyFile);
+        InputStream defaultPropertyStream = getClassLoader().getResourceAsStream(defaultPropertyFile);
 
         if (defaultPropertyStream != null) {
             defaultProperties = new Properties();
@@ -110,7 +110,7 @@ public class PropertyUtil {
         }
 
         // Load the property file
-        InputStream propertyStream = PropertyUtil.class.getClassLoader().getResourceAsStream(propertyFile);
+        InputStream propertyStream = getClassLoader().getResourceAsStream(propertyFile);
 
         if (propertyStream != null) {
             // Add default properties if they were found
@@ -140,5 +140,13 @@ public class PropertyUtil {
         }
 
         return properties;
+    }
+
+    private static ClassLoader getClassLoader() {
+        ClassLoader classLoader = PropertyUtil.class.getClassLoader();
+        if (classLoader == null) {
+            classLoader = ClassLoader.getSystemClassLoader();
+        }
+        return classLoader;
     }
 }

--- a/src/main/java/net/jacobpeterson/alpaca/util/properties/PropertyUtil.java
+++ b/src/main/java/net/jacobpeterson/alpaca/util/properties/PropertyUtil.java
@@ -85,7 +85,7 @@ public class PropertyUtil {
 
         // Load the default property file if exists
         Properties defaultProperties = null;
-        InputStream defaultPropertyStream = ClassLoader.getSystemClassLoader().getResourceAsStream(defaultPropertyFile);
+        InputStream defaultPropertyStream = PropertyUtil.class.getClassLoader().getResourceAsStream(defaultPropertyFile);
 
         if (defaultPropertyStream != null) {
             defaultProperties = new Properties();
@@ -110,7 +110,7 @@ public class PropertyUtil {
         }
 
         // Load the property file
-        InputStream propertyStream = ClassLoader.getSystemClassLoader().getResourceAsStream(propertyFile);
+        InputStream propertyStream = PropertyUtil.class.getClassLoader().getResourceAsStream(propertyFile);
 
         if (propertyStream != null) {
             // Add default properties if they were found

